recipe/meta.yaml

```diff
@@ -1,60 +1,77 @@
 {% set name = "hmmlearn" %}
-{% set version = "0.2.8" %}
+{% set version = "0.3.2" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 696931e3dce6801cc9b78c629f543061dd7eb67a71155b03d1b39ea172a2a5f9
+  sha256: edaf485fdb1ea88da9ac642b2006c63d9950dd15d4d132f7205305d383e6f745
 
 build:
-  number: 1
-  skip: true  # [py<35]
+  number: 0
+  skip: true  # [py<38]
   script:
-    - {{ PYTHON }} setup.py build_ext
-    - {{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv  # [unix]
-    - {{ PYTHON }} setup.py install --single-version-externally-managed --record record.txt  # [win]
+    - {{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-build-isolation -vv
+  missing_dso_whitelist:  # [s390x]
+    - '$RPATH/ld64.so.1'  # [s390x]
 
 requirements:
   build:
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
-    - numpy                                  # [build_platform != target_platform]
-    - pybind11                               # [build_platform != target_platform]
     - {{ compiler('c') }}
     - {{ compiler("cxx") }}
   host:
     - pip
     - python
-    - numpy
     - pybind11 >=2.6
     - setuptools
-    - setuptools_scm
+    - setuptools_scm >=3.3
+    - pytest
+    - wheel
   run:
     - python
-    - scikit-learn >=0.16
-    - {{ pin_compatible('numpy') }}
+    - scikit-learn >=0.16,!=0.22.0
+    - numpy >=1.10
     - scipy >=0.19.0
-    - setuptools
-    - matplotlib-base >=1.1.1
+    # mkl variant pulls in intel-openmp which conflicts with llvm-openmp. i.e. force to use openblas variant of numpy, scipy, etc.
+    - blas=*=openblas  # [osx and x86_64]
 
 test:
+  source_files:
+    - .
   imports:
     - hmmlearn
     - hmmlearn.tests
+  commands:
+    - pip check
+    # Test suite requires package to be built in place.
+    - pip install -e .
```

Comment:
In this way we are installing from `pypi` all the dependencies, and then build the package using those.

Why do the tests need a rebuild of the package? What do you mean by `editable`?

Reply:
They run the tests with
```
python -I -mpytest --pyargs hmmlearn.tests
```
https://github.com/hmmlearn/hmmlearn/blob/33b1a916a11efb5c569abfc9705b7ae030e81c31/.github/workflows/build.yml#L36C28-L36C70

Reply:
The tests were unable to run, complaining that they can't import `_hmmc`. See error below:

```
+ pytest lib/hmmlearn/tests -v
ImportError while loading conftest '/opt/miniconda3/conda-bld/hmmlearn_1718041400404/test_tmp/lib/hmmlearn/tests/conftest.py'.
lib/hmmlearn/tests/__init__.py:9: in <module>
    from hmmlearn.base import DECODER_ALGORITHMS
lib/hmmlearn/base.py:12: in <module>
    from . import _hmmc, _kl_divergence as _kl, _utils
E   ImportError: cannot import name '_hmmc' from 'hmmlearn' (/opt/miniconda3/conda-bld/hmmlearn_1718041400404/test_tmp/lib/hmmlearn/__init__.py)
```

This is what I was able to find on the issue: https://github.com/hmmlearn/hmmlearn/issues/539#issuecomment-1871284838

As for editable, they use `python setup.py develop`, which is equivalent to `pip install . -e`. It's basically `setuptools` development mode. More info here: https://setuptools.pypa.io/en/latest/userguide/development_mode.html

Reply:
All that said, I didn't realize it was pulling the deps from pypi. Barring using a completely different method to get the tests running, adding `--no-deps` would ensure the deps are pulled from defaults, right?

Reply:
Disregard all of that. The command you found works perfectly. Thanks!